libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/utils/test/scheduledMailRecipients.test.ts

```diff
@@ -0,0 +1,131 @@
+// (C) 2023 GoodData Corporation
+
+import { splitScheduledEmailRecipients, uniqueScheduledEmailRecipients } from "../scheduledMailRecipients";
+
+const scheduleEmailExistingRecipient = {
+    user: {
+        email: "test@gooddata.com",
+        firstName: "First name",
+        fullName: "Full name",
+        lastName: "Last name",
+        login: "test@gooddata.com",
+        ref: {
+            uri: "/gdc/account/profile/123456",
+        },
+    },
+};
+
+const scheduleEmailExternalRecipientSameExisting = {
+    email: "test@gooddata.com",
+};
+
+const scheduleEmailExternalRecipient = {
+    email: "external@gooddata.com",
+};
+
+const scheduleEmailExternalRecipientWithUnsupportedDelimiter = {
+    email: "external@gooddata.com|new_external@gooddata.com",
+};
+
+const scheduleEmailExternalRecipientWithCommaDelimiter = {
+    email: "external@gooddata.com,external1@gooddata.com",
+};
+
+const scheduleEmailExternalRecipientWithSemicolonDelimiter = {
+    email: "external@gooddata.com; external1@gooddata.com",
+};
+
+const scheduleEmailExternalRecipientWithSpaceDelimiter = {
+    email: "external@gooddata.com external1@gooddata.com",
+};
+
+const scheduleEmailExternalRecipientWithMultipleDelimiter = {
+    email: "external@gooddata.com,external1@gooddata.com;  external2@gooddata.com",
+};
+
+describe("schedule email recipients utils", () => {
+    describe("split schedule email recipients", () => {
+        it("with unsupported delimiter", () => {
+            const recipients = [
+                scheduleEmailExistingRecipient,
+                scheduleEmailExternalRecipientWithUnsupportedDelimiter,
+            ];
+            const result = splitScheduledEmailRecipients(recipients);
+            expect(result).toHaveLength(2);
```

Comment:
For all these expects, I would rather use snapshot, i.e., `expect(result).toMatchSnapshot();`.
This way you can be sure that regex split the emails correctly. For example did not cut first letter of email address, trimmed the white spaces around it, etc. With this condition you only know that there are 2 emails parsed and someone can change the regex in future to mess up the parsed emails and he would not know he did a mistake.